components/hero-banner.php

```diff
@@ -162,139 +162,409 @@ function mm_hero_banner_shortcode( $atts, $content = null, $tag ) {
+            ),
+            array(
+                'type' => 'textfield',
+                'heading' => __( 'Button Text', 'mm-components' ),
+                'param_name' => 'button_text',
+            ),
+            array(
+                'type' => 'dropdown',
+                'heading' => __( 'Button Style', 'mm-components' ),
+                'param_name' => 'button_style',
+                'value' => array(
+                    __( 'Default (solid)', 'mm-components ') => 'default',
+                    __( 'Ghost (transparent background, white border)', 'mm-components ') => 'ghost',
+                ),
+            ),
+            array(
+                'type' => 'dropdown',
+                'heading' => __( 'Button Color', 'mm-components' ),
+                'param_name' => 'button_color',
+                'value' => array(
+                    __( 'Default', 'mm-components ') => 'default',
+                    __( 'Pink', 'mm-components ') => 'pink',
+                    __( 'White', 'mm-components ') => 'white',
+                    __( 'Gray', 'mm-components ') => 'gray',
+                ),
+                'dependency' => array(
+                    'element' => 'button_style',
+                    'value' => array(
+                        'ghost',
+                    ),
+                ),
+            ),
+            array(
+                'type' => 'textarea_raw_html',
+                'heading' => __( 'Secondary Call to Action', 'mm-components' ),
+                'param_name' => 'secondary_cta',
+                'description' => __( 'Outputs below the main button, can include HTML markup.', 'mm-components' ),
+            ),
+        ),
+    ) );
+}
+
+add_action( 'widgets_init', 'mm_components_register_hero_banner_widget' );
+/**
+ * Register the widget.
+ *
+ * @since  1.0.0
+ */
+function mm_components_register_hero_banner_widget() {
+    register_widget( 'mm_hero_banner_widget' );
+}
+/**
+ * Hero Banner Widget.
+ *
+ * @since  1.0.0
+ */
+class Mm_hero_banner_Widget extends Mm_Components_Widget {
+    /**
+     * Global options for this widget.
+     *
+     * @since  1.0.0
+     */
+    protected $options;
+    /**
+     * Initialize an instance of the widget.
+     *
+     * @since  1.0.0
+     */
+    public function __construct() {
+        // Set up the options to pass to the WP_Widget constructor.
+        $this->options = array(
+            'classname'   => 'mm-hero-banner-strip',
+            'description' => __( 'A Hero Banner', 'mm-components' ),
+        );
+        parent::__construct(
+            'mm_hero_banner_widget',
+            __( 'Mm Hero Banner', 'mm-components' ),
+            $this->options
+        );
+    }
+    /**
+     * Output the widget.
+     *
+     * @since  1.0.0
+     *
+     * @param  array  $args      The global options for the widget.
+     * @param  array  $instance  The options for the widget instance.
+     */
+    public function widget( $args, $instance ) {
```

Comment:
Check that when a new instance of the widget is first added to a widget area in the Customizer, this doesn't generate errors. I think you need to define defaults here and use wp_parse_args. Check the Blockquote widget as an example.